api/client/options.proto

```diff
@@ -14,6 +14,10 @@ message RequestOptions {
   google.protobuf.UInt32Value request_body_size = 3 [(validate.rules).uint32 = {lte: 4194304}];
 }
 
+message RequestSource {
+  google.protobuf.StringValue uri = 19; // [(validate.rules).string.uri = true];
```

Comment:
Curious, why did we choose to wrap the uri in a message? Do we expect more qualifiers for a request source?